### ISSUE TITLE
Refund implementation

### ISSUE THREAD

**Issue by jjboxxed** [nozzlegear/ShopifySharp#400]
I need to issue refunds via API. Happy to implement if you give me some pointers.

Thanks

**Comment by nozzlegear** [comment]
Hey @jjboxxed, I'm happy to hear you want to contribute! The first step would be to find Shopify's documentation for whatever you want to implement. Quickest way to do that would be a Google search like `shopify refund api reference`. 

After that, you should create a new class named e.g. `Refund` if it doesn't already exist and then start adding the new properties. Shopify snake_cases their properties but in C# we like to use PascalCase. In practice that means each property on your class will need a `JsonAttribute` that translates the snake case name to pascal case:

```cs
public class MyNewClass
{
    [JsonProperty("shopify_property_name")]
    public string ShopifyPropertyName { get; set; }
}
```

If you have int, long, decimal, datetime, etc. properties, they should always be made nullable, e.g. `int?` `long?` `decimal?` `DateTime?`. 

With the class done, you'll need to create the new service, with a method for each endpoint on Shopify's documentation page. You can check out the [OrderService](https://github.com/nozzlegear/ShopifySharp/blob/8b0305dcb9fa7e041d22d64c1502255fd58b24b6/ShopifySharp/Services/Order/OrderService.cs) for an example. 

You should write some tests for the new service too, and again you can use the [Order tests](https://github.com/nozzlegear/ShopifySharp/blob/master/ShopifySharp.Tests/Order_Tests.cs) for inspiration. They don't have to be elaborate, they should just make sure doing things like listing, creating, counting, getting, etc. work. Although for the refund, it's probably hard to write a test for creating and updating refunds so it'd be okay to skip them. 

For your specific case, you'll need to be careful. I think there's more than one type of refund in Shopify's API: a refund for an order, and a refund for application billing. You'll want to rename the new class/service appropriately (e.g. ApplicationRefund or OrderRefund). 

Let me know if you have any other questions or need more help! The PR doesn't have to be perfect, I can review it once you're ready to merge it in and if there are issues we can work through them together.

**Comment by jjboxxed** [comment]
Great, thanks. I'll get on it as soon as time allows.

**Comment by PaulThomasRowe** [comment]
I've made a pull request that should resolve this issue. #436

**Comment by nozzlegear** [comment]
This was published in version 4.25.0 on nuget. Thanks for opening the issue, and thanks to @PaulThomasRowe for implementing! 🎉
